CMakeLists.txt

```diff
@@ -204,7 +204,7 @@ include_directories(SYSTEM
   ${OPENGL_INCLUDE_DIR}
   ${PYTHON_INCLUDE_PATH}
 )
-include_directories(src ${catkin_INCLUDE_DIRS})
+include_directories(src SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
I don't want to use `SYSTEM`, it's known to cause all sorts of problems with the include order on some OS's.

Please use the diagnostics pragma's to disable the particular warnings when including the problem headers. See:

https://gcc.gnu.org/onlinedocs/gcc/Diagnostic-Pragmas.html

Here's an example of it in use:

https://github.com/ros2/rmw_connext/blob/master/rmw_connext_cpp/src/functions.cpp#L26-L40